### ISSUE TITLE
CDK Hotswap Feature : AWS::SQS::Queue Attribute Support

### ISSUE THREAD

**Issue by naedx** [aws/aws-cdk#30841]
### Describe the bug

A project with the L3 construct [LambdaToSqsToLambda](https://constructs.dev/packages/@aws-solutions-constructs/aws-lambda-sqs-lambda/v/2.61.0?lang=typescript) falls back to full deployments everytime because of the SQS Queue

### Expected Behavior

A hotswap is perfomed.

### Current Behavior

A full deployment is performed.

```

amplify-xxxx-xxxx-sandbox-xxxx: deploying... [1/1]
Could not perform a hotswap deployment, because the CloudFormation template could not be resolved: We don't support attributes of the 'AWS::SQS::Queue' resource. This is a CDK limitation. Please report it at https://github.com/aws/aws-cdk/issues/new/choose
Falling back to doing a full deployment
amplify-xxxx-xxxx-sandbox-xxxx: updating stack...

```

### Reproduction Steps

1. Create an Amplify project
2. Add the L3 construct [LambdaToSqsToLambda](https://constructs.dev/packages/@aws-solutions-constructs/aws-lambda-sqs-lambda/v/2.61.0?lang=typescript)
3. Deploy.

### CDK CLI Version

2.147.0 (build 3338fc0)

### Node.js Version

20.9.0

### OS

macOS 14.5, arm64

### Language

TypeScript

### Language Version

5.5.2

**Comment by khushail** [comment]
Hey @naedx , thanks for reaching out. Could you please check if your issue is similar to this existing one -
https://github.com/aws/aws-cdk/issues/23533  and several others mentioned under it ?

**Comment by naedx** [comment]
I'll take a look at it and let you know shortly

**Comment by naedx** [comment]
Hi @khushail , I think it's related. From the thread I gathered that the cross stack references are linked to the issue occuring. 

My SQS references a lambda which references a stack with S3 resources and another lambda which references a stack with AppSync/DynamoDb resources. The lambdas are declared in the stack of the resources that they reference.

I tried removing the references between the lambda and the S3 and AppSync/DynamoDB resources. Hotswap still doesn't work.

I then moved the SQS and both lambdas to the same stack and this allows hotswap to work again. 

This sheds some light on the issue but it's not a workable solution for me because I cannot re-add access to my S3, AppSync/DynamoDB resources without running into a circular dependency issues.
